core/services/ocr2/plugins/directrequestocr/codec.go

```diff
@@ -0,0 +1,97 @@
+package directrequestocr
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/coreth/accounts/abi"
+	"github.com/pkg/errors"
+)
+
+type reportCodec struct {
+	reportTypes abi.Arguments
+}
+
+func getReportTypes() (abi.Arguments, error) {
+	bytes32ArrType, err := abi.NewType("bytes32[]", "", []abi.ArgumentMarshaling{})
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create an ABI type object for bytes32[]")
+	}
+	bytesArrType, err := abi.NewType("bytes[]", "", []abi.ArgumentMarshaling{})
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create an ABI type object for bytes[]")
+	}
+	return abi.Arguments([]abi.Argument{
+		{Name: "ids", Type: bytes32ArrType},
+		{Name: "results", Type: bytesArrType},
+		{Name: "errors", Type: bytesArrType},
+	}), nil
+}
+
+func NewReportCodec() (*reportCodec, error) {
+	reportTypes, err := getReportTypes()
+	if err != nil {
+		return nil, err
+	}
+	return &reportCodec{
+		reportTypes: reportTypes,
+	}, nil
+}
+
+func sliceToByte32(slice []byte) [32]byte {
+	return *(*[32]byte)(slice)
```

Comment:
This does not look safe. The data under the slice can be changed. I would rather allocate `[32]byte` and `copy`.